src/fnft_nsep.c

```diff
@@ -112,28 +111,17 @@ INT fnft_nsep(const UINT D, COMPLEX const * const q,
     if (opts_ptr->filtering != fnft_nsep_filt_NONE && main_spec == NULL && aux_spec != NULL)
         return E_INVALID_ARGUMENT(main_spec. Filtering of the auxiliary spectrum is not possible if the main spectrum is not computed.);
       
-        
-        if (CABS(q[0]) == 0){
-            WARN("First sample of signal q is 0. Not possible to determine if signal is quasi-periodic. Continuing assuming signal is periodic. Computed spectrum maybe incorrect.");
-        }else
-            m_q = q[D-1]/q[0];
-        
-        if (FABS(1-CABS(m_q)) > 1e-2)
-            WARN("Signal q may not be (quasi-)periodic. abs(1-abs(q[D-1]/q[0]))>1e-2.");
-        
-        REAL Lam_shift = CARG(m_q)/(-2*(T[1] - T[0]));
-        const REAL eps_t = (T[1] - T[0])/(D - 1);
-        D_effective = D-1;
-        q_preprocessed = malloc(D_effective * sizeof(COMPLEX));
+        REAL Lam_shift = CARG(CEXP(I*phase_shift))/(-2*(T[1] - T[0]));
```

Comment:
Could `CARG(CEXP(I*phase_shift))` not be replaced with `phase_shift`?

Reply:
I am actually not sure about this. I had put that in incorrectly thinking that phase_shift needs to be between -PI and PI. However if you look at the reference it appears this need not be the case. Bijectivity of the  mapping from the phase_shift to Lam_shift  is lost if phase_shift is restricted to -PI to PI. I think the solution would be to indeed just replace CARG(CEXP(I*phase_shift)) with phase_shift and add a more elaborate note mentioning that it should be the unwrapped phase_shift. Please let me know your thoughts.

Reply:
I see. If you look down below, you indeed see that additional multiple's of pi mess up the factors. Maybe mentioned and add a check at the beginning of the function?

Reply:
I did some checks in MATLAB and now realize that the additional multiples of pi will not mess things up because the `phase_shift` is used to obtain an intermediate signal which is also periodic. The `Lam_shift` is just relative to the spectrum of this intermediate signal.

Reply:
Shall I mark this as resolved?

Reply:
Yes, please.